WalletWasabi/Blockchain/Transactions/SmartTransaction.cs

```diff
@@ -99,7 +99,7 @@ public SmartTransaction(Transaction transaction, Height height, uint256? blockHa
 		/// * Explicitly by using a nSequence &lt; (0xffffffff - 1) or,
 		/// * Implicitly in case one of its unconfirmed ancestors are replaceable
 		/// </summary>
-		public bool IsRBF => !Confirmed && (Transaction.RBF || IsReplacement);
+		public bool IsRBF => !Confirmed && (Transaction.RBF || IsReplacement || WalletInputs.Any(x => x.IsReplaceable()));
```

Comment:
I'm not sure it does what it want you to do.  

`WalletInputs` are set at the end of the transaction processing with `Coins.Spend(coin, tx);` and `IsRBF` is used at the beginning of the transaction processing: `var isReplacementTx = doubleSpends.Any(x => x.IsReplaceable());`

There are other places of course when it works well, but in this case it doesn't

Reply:
I am wrong. The transaction processing indeed sets the wallet inputs at the end, but the `IsRBF` isn't used for the current transaction, rather it's checked for previous transactions where it's already set at previous processing.